src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -49,39 +51,43 @@
      * The human readable name of the plugin as to be used in descriptions and
      * similar things.
      *
-     * @return The plugin name
+     * @return The plugin name, or an empty string if unknown
```

Comment:
Not on the annotation (because then there is no way to tell if it was specified or not). For the `PluginContainer` there is `getDisplayName` which returns the plugin ID if the name is not present (I was considering doing that on `getName` there directly, but then you'd have no way to tell if the name is present or not... not sure about that yet).